### PR TITLE
Slider component

### PR DESCRIPTION
**Pull request by guilhermemntt**
This PR adds a slider component to ``pi-ui`` library, both single and double handled, in order to suit several contexts. This work was inspired by different implementations and has several specific adaptations to meet the needs of React Decred's projects.

Closes https://github.com/decred/pi-ui/issues/307.

### CODE REVIEW

File: .eslintrc
Comment (amassarwi) [inline .eslintrc:31]: Can you please explain why you added this here ?

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:13]: These should be in separate file - create `hooks.js` in the same dir

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:17]: Define new const instead of using array everywhere: `const leftHnadle = handle[0]`

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:59]: define consts instead of arr[index]

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:139]: warnings

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:151]: warnings

File: src/components/Slider/SliderHandle.jsx
Comment (amassarwi) [inline src/components/Slider/SliderHandle.jsx:12]: same here hook should be in `hooks.js`

File: src/components/Slider/styles.css
Comment (amassarwi) [inline src/components/Slider/styles.css:1]: please don't use `px` units, we use `rem` units all over the app.

we defined our root font for be 10px, so convert any px value to rem using `1rem = 10px`

**Review by amassarwi — CHANGES_REQUESTED** [review]
Thanks @guilhermemntt tested and looks neat, left some comments

File: .eslintrc
Comment (tiagoalvesdulce) [reply .eslintrc:31]: This rule enforces the use of the correct prefix for event handling events. IMO it's too strict.

**Review by tiagoalvesdulce — CHANGES_REQUESTED** [review]
Besides amass's suggestions I would like to point out that you are overusing `useCallback()`. It does nothing good when you add the functions as event listeners, in fact it makes the code slower because of the extra comparison and prevents those functions to be properly garbage collected. Also, you are adding non-primitive types to the deps list and they will not have the same reference overtime. For more info read [this article](https://kentcdodds.com/blog/usememo-and-usecallback) but a good rule of thumb is: only use memoization for expensive calculations or to ensure referential equality. Please reconsider your `useCallback` use.

**Comment by guilhermemntt** [comment]
Thanks @amassarwi! eslint was pointing out errors due to ``react/jsx-handler-names`` [default value](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-handler-names.md#rule-options).I thinks it's too strict as well, but I decided to setting this flag to 1 to avoid errors and indicate a warning instead of an error. Now, it's set to 0 explicitly (no warnings and no erros).
About the ``useCallback`` use, you're totally right. ``handles`` is an array and ``onChange`` a listener, so the implementation does not make any sense. Also, ``value`` is changed constantly, so I reconsidered the use of this hook to avoid unnecessary memorization (even when using ``useEffect``, for example). Thanks @tiagoalvesdulce for the important consideration.
I think it's ready for another review.

**Review by amassarwi — APPROVED** [review]
Code looks good to me, good job.
I'd suggest we open a PR in decrediton using this branch to test how it plays, we should merge this one only when we are sure it works as expected there.

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:38]: as you access `handleHooks[1]` & `handleHooks[0]` please consider defining a const for each one of them maybe with a bit more meaningful name

**Comment by guilhermemntt** [comment]
Thanks @amassarwi. So, I'll continue this work by solving https://github.com/decred/decrediton/issues/2874.

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:204]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:206]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:211]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:213]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:171]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:181]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:135]: extra empty line

File: src/components/Slider/Slider.jsx
Comment (amassarwi) [inline src/components/Slider/Slider.jsx:26]: extra empty line

File: src/components/Slider/hooks.js
Comment (amassarwi) [inline src/components/Slider/hooks.js:52]: too many empty lines in this `change` function.

File: src/components/Slider/helpers.js
Comment (amassarwi) [inline src/components/Slider/helpers.js:31]: extra empty line

**Review by amassarwi — COMMENTED** [review]
LGTM, just some cleanup comments.
Please follow existing code style and guidelines.

**Review by amassarwi — APPROVED** [review]
ACK, needs @tiagoalvesdulce approval tho.

**Comment by guilhermemntt** [comment]
Thanks @amassarwi.

**Review by tiagoalvesdulce — APPROVED** [review]
Tested and works great! LGTM

**Pull request merged**
